app/src/org/commcare/android/adapters/HomeScreenAdapter.java

```diff
@@ -0,0 +1,186 @@
+package org.commcare.android.adapters;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+
+import org.commcare.android.view.SquareButtonWithNotification;
+import org.commcare.dalvik.R;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+
+/**
+ * Created by dancluna on 3/19/15.
```

Comment:
It would really help if you gave some context as to what this class did, even if it is just saying 'sets up home screen buttons and gives accessors for setting their visibility and listeners'.